### PR TITLE
Update antlr to 4.13 for 33% speed up on some inputs

### PR DESCRIPTION
**Pull request by siliconvoodoo**
This is a suggestion of a possible way to update antlr.
This branch could use an intervention from an o3de maintainer to go and update the [o3de/antlr4](https://github.com/o3de/antlr4) fork and put that reference in the CMakeList in a supplementary commit on top of this branch.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -13,8 +13,8 @@ FetchContent_Declare(
 # Antlr4 external dependency
 FetchContent_Declare(
   antlr4
-  GIT_REPOSITORY https://github.com/o3de/antlr4.git
-  GIT_TAG o3de-4.9.3
+  GIT_REPOSITORY https://github.com/antlr/antlr4/
+  GIT_TAG 4.13.2
```
Comment (siliconvoodoo) [inline src/CMakeLists.txt]: this is the point that would need a final commit to polish back to o3de fork.

**Comment by galibzon** [comment]
Awesome. I'll take a look at o3de/antlr and add a tag for 4.13.2.

**Review by galibzon — APPROVED** [review]

**Comment by galibzon** [comment]
`o3de/antlr` now has the `o3de-4.13.2` branch: https://github.com/o3de/antlr4/tree/o3de-4.13.2

**Review by moudgils — APPROVED** [review]
Nice optimization

**Comment by siliconvoodoo** [comment]
> `o3de/antlr` now has the `o3de-4.13.2` branch: https://github.com/o3de/antlr4/tree/o3de-4.13.2

splendid. ready to merge now.
at next o3de release let's not forget to make the 1.8.20 3p package :p

**Pull request merged**